k9mail-library/src/main/java/com/fsck/k9/mail/transport/SmtpTransport.java

```diff
@@ -308,11 +308,19 @@ public void open() throws MessagingException {
                 authExternalSupported = saslMech.contains("EXTERNAL");
             }
             if (extensions.containsKey("SIZE")) {
-                try {
-                    mLargestAcceptableMessage = Integer.parseInt(extensions.get("SIZE"));
-                } catch (Exception e) {
-                    if (K9MailLib.isDebug() && DEBUG_PROTOCOL_SMTP) {
-                        Log.d(LOG_TAG, "Tried to parse " + extensions.get("SIZE") + " and get an int", e);
+                String sizeValue = extensions.get("SIZE");
+                /*
+                    An empty value is valid and indicates no limit :
+
+                    "**If** a numeric parameter..." - RFC 1870 4.
+                 */
+                if (sizeValue != null && sizeValue != "") {
+                    try {
+                        mLargestAcceptableMessage = Integer.parseInt(extensions.get("SIZE"));
```

Comment:
Please reuse `sizeValue` instead of calling `extensions.get("SIZE")` a second time.